File: trunk/ShadowFrameworkLite20/src/shadow/image/bitmaps/SFFunctionRandomizer.java
package shadow.image.bitmaps;

import shadow.math.SFRandomizer;

public class SFFunctionRandomizer implements SFBitmapFunction{

	private SFRandomizer randomizer;

	public SFFunctionRandomizer(int seed) {
		super();
		this.randomizer=new SFRandomizer(seed);
	}
	
	@Override
	public float getValue(float u, float v) {
		return randomizer.randomFloat();
	}
	
	@Override
	public void destroy() {
		
	}
	
	@Override
	public void init() {
		
	}
}

File: ShadowFrameworkJS/src/shadow/image/bitmaps/SFFunctionRandomizer.js


function SFFunctionRandomizer(seed){
	this.randomizer=new SFRandomizer(seed);
}


SFFunctionRandomizer.prototype["getValue"]=function(u,v){
		return this.randomizer.randomFloat();
};

SFFunctionRandomizer.prototype["init"]=function(){
	// do nothing
};

SFFunctionRandomizer.prototype["destroy"]=function(){
	// do nothing
};





